nixos/modules/services/hardware/fwupd.nix

```diff
@@ -0,0 +1,90 @@
+# fwupd daemon.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.fwupd;
+  originalEtc =
+    let
+      isRegular = v: v == "regular";
+      listFiles = d: builtins.attrNames (filterAttrs (const isRegular) (builtins.readDir d));
```

Comment:
Please, not this!  It causes **building** during **evaluation** of your system, which breaks various assumptions, like Hydra rebuilding stdenv when it's only supposed to get a list of jobs.  Especially as this service is enabled by default.

Reply:
BTW, we apparently had some other places like this, but they most likely haven't been tracked down yet.  https://github.com/NixOS/nixpkgs/issues/29774

Reply:
Agreed, this should be fixed or reverted. And why is the service enabled by default?

Reply:
I have disabled it by default in 2f1a818d00f957f3102c0b412864c63b6e3e7447

Reply:
`fwupd` increases the size of the smallContainer from 400 to 800 MB: https://hydra.nixos.org/job/nixos/trunk-combined/nixos.closures.smallContainer.x86_64-linux#tabs-charts

Reply:
Yeah, not suitable to be on by default in this state.

Reply:
Sorry about enabling it by default, that was not intended.

Is there any better way to install the files to etc other than hardcoding the list of files?

Reply:
I don't know of any.  You often don't even need to put anything into `/etc/` and you just pass paths to some /nix/store/... as config.  But I'm not really good at nixos module best practices.

Reply:
> Is there any better way to install the files to etc other than hardcoding the list of files?

I see three options:
- hardcode the list of files
- write a custom activation script
- extend the existing activation script: https://github.com/NixOS/nixpkgs/tree/master/nixos/modules/system/etc/

I am not sure which is better: the first is fine if fwupd is the only package that needs it (you may list the files in a passthru attribute of the `fwupd` definition to keep them closer to the source), the second is hard to justify, and the third is more work and may be hard to get right, but it the right approach to support more dynamic /etc.